### PR TITLE
Separate backup

### PR DESCRIPTION
**Pull request by benjaminjb**
* separates out the backup logic from the cobra command
* adds a basic test to the backup command

Note: `backup` updates the cluster spec, which updates the job pod, so this command is best tested with an e2e test with a running PGO.

(The "cluster spec" is also unstructured, another reason why running an e2e test is the best way to ensure confidence.)

Issue: PGO-663

### CODE REVIEW

File: testing/kuttl/e2e/backup/README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
 * Check the annotation on the cluster
 * No backup occurs
 
-(6) 10-11
+(6) 10-12
 * Update the spec through KUTTL, changing the ownership of that field
 * Call the backup CLI with different flags, and see a conflict
+* Call the backup CLI with different flags and the `--force-conflict` flag, see no conflict
```
Comment (tony-landreth) [inline testing/kuttl/e2e/backup/README.md:27]:
```suggestion
* Call the backup CLI with different flags and the `--force-conflicts` flag, see no conflict
```

File: internal/cmd/backup.go [hunk 2]
```diff
@@ -185,3 +141,67 @@ func (config pgBackRestBackup) modifyIntent(
 
 	return nil
 }
+
+func (backup pgBackRestBackup) Run(client dynamic.NamespaceableResourceInterface,
+	cmd *cobra.Command,
+	clusterName string) error {
+
+	var (
```
Comment (benjaminjb) [inline internal/cmd/backup.go:149]: Remove

File: internal/cmd/backup.go [hunk 3]
```diff
@@ -79,88 +79,45 @@ postgresclusters/hippo backup initiated`)
...
-		)
-
-		if err != nil {
-			if apierrors.IsConflict(err) {
-				cmd.Printf("SUGGESTION: The --force-conflicts flag may help in performing this operation.")
-			}
-			cmd.Printf("\nError requesting update: %s\n", err)
-			return err
+		if err == nil {
+			// Consider a `--wait` flag
```
Comment (tony-landreth) [inline internal/cmd/backup.go:95]: Not sure I understand this comment.

Comment (benjaminjb) [reply internal/cmd/backup.go:95]: When we request a backup, we initiate the process, but the user would then have to check if the backup job had a completed pod or check the spec to see that the most recent backup was complate. Would it be useful to wait until the backup is done?

Comment (benjaminjb) [reply internal/cmd/backup.go:95]: I should add that ^ context to the comment.

File: internal/cmd/backup.go [hunk 4]
```diff
@@ -79,88 +79,47 @@ postgresclusters/hippo backup initiated`)
...
+type pgBackRestBackupArgs struct {
+	ClusterName    string
+	ForceConflicts bool
 	Options        []string
 	RepoName       string
-	ForceConflicts bool
 }
 
-func (config pgBackRestBackup) modifyIntent(
+func (backup pgBackRestBackupArgs) modifyIntent(
```
Comment (tony-landreth) [inline internal/cmd/backup.go:114]: Should `modifyIntent` methods be defined on Unstructured structs?

**Review by tony-landreth — APPROVED** [review]

**Pull request merged**